erpnext/stock/doctype/stock_entry/stock_entry.py

```diff
@@ -692,18 +692,19 @@ def update_stock_ledger(self):
 		self.make_sl_entries(sl_entries, self.amended_from and 'Yes' or 'No')
 
 	def validate_mandatory_batch_number(self):
-		# to validate the stock items batch number is present or not
+		"""validate the stock items batch number is present or not"""
 		items_without_batch_no = []
 		MANDATE_MSG = 'Batch number is mandatory for Item at : <br><b>'
-		item_list = self.get("items")
-		
+		item_list = self.get("items")		
+		# iterate through all row of stock entry item and check whether batch no for the item is present or not by querying to tabItem table
 		for index in range(len(item_list)):
-			item_code_list = self.items[index].item_code
+			item_code = self.items[index].item_code
```

Comment:
Suggested change: if this is a single item, then just use item_code. Avoid using list for a string variable